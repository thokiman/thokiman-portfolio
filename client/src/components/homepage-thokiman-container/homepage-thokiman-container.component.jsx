import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectIsSideBarHidden } from "../../redux/header/header.selectors";
import {
  HomepageThokimanBodyContainer,
  HomepageThokimanContainer,
  HomepageThokimanContentContainer,
  HomepageThokimanHeadContainer,
  HomepageThokimanHeadOutline,
  HomepageWhiteThokimanSkeleton,
} from "./homepage-thokiman-container.styles";

export const HomePageThokimanContentContainer = ({ isSideBarHidden }) => {
  return (
    <HomepageThokimanContentContainer>
      <HomepageThokimanContainer $issidebarhidden={isSideBarHidden}>
        <HomepageThokimanHeadContainer>
          <HomepageThokimanHeadOutline />
        </HomepageThokimanHeadContainer>
        <HomepageThokimanBodyContainer>
          <HomepageWhiteThokimanSkeleton />
        </HomepageThokimanBodyContainer>
      </HomepageThokimanContainer>
    </HomepageThokimanContentContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(HomePageThokimanContentContainer);
