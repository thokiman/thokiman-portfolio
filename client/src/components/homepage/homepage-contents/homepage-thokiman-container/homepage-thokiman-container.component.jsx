import gsap from 'gsap';
import React, { useEffect, useRef } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import {
  selectIsHeaderMorphingActive,
  selectIsSideBarActive,
} from '../../../../redux/header/header.selectors';
import { selectRockWaveImage } from '../../../../redux/homepage/homepage.selectors';
import {
  HomepageThokimanBodyContainer,
  HomepageThokimanContainer,
  HomepageThokimanContentContainer,
  HomepageThokimanHeadContainer,
  HomepageThokimanHeadOutline,
  HomepageWhiteThokimanSkeleton,
} from './homepage-thokiman-container.styles';

export const HomePageThokimanContentContainer = ({
  isSideBarActive,
  rockWaveImage,
  isHeaderMorphingActive,
  toggleThokimanSkeleton,
  toggleThokimanOutline,
}) => {
  const homepageThokimanHeadOutlineRef = useRef();
  const homepageWhiteThokimanSkeletonRef = useRef();
  const homepageThokimanHeadOutlineTl = useRef();
  const homepageWhiteThokimanSkeletonTl = useRef();

  useEffect(() => {
    if (isHeaderMorphingActive) {
      homepageThokimanHeadOutlineTl.current = gsap
        .timeline()
        .to(homepageThokimanHeadOutlineRef.current, {
          yPercent: 40,
          duration: 6,
          yoyo: true,
          yoyoEase: true,
          repeat: 1,
          repeatDelay: 0.2,
        });
      homepageWhiteThokimanSkeletonTl.current = gsap
        .timeline()
        .to(homepageWhiteThokimanSkeletonRef.current, {
          yPercent: 60,
          duration: 6,
          yoyo: true,
          yoyoEase: true,
          repeat: 1,
          repeatDelay: 0.4,
        });
      homepageThokimanHeadOutlineTl.current.play();
      homepageWhiteThokimanSkeletonTl.current.play();
    }
  });
  return (
    <HomepageThokimanContentContainer
      $rockWaveImage={rockWaveImage}
      className='homepage-thokiman-content-container'
    >
      <HomepageThokimanContainer $issidebaractive={isSideBarActive}>
        <HomepageThokimanHeadContainer>
          <HomepageThokimanHeadOutline ref={homepageThokimanHeadOutlineRef} />
        </HomepageThokimanHeadContainer>
        <HomepageThokimanBodyContainer>
          <HomepageWhiteThokimanSkeleton
            ref={homepageWhiteThokimanSkeletonRef}
          />
        </HomepageThokimanBodyContainer>
      </HomepageThokimanContainer>
    </HomepageThokimanContentContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isHeaderMorphingActive: selectIsHeaderMorphingActive,
  isSideBarActive: selectIsSideBarActive,
  rockWaveImage: selectRockWaveImage,
});

export default connect(mapStateToProps)(HomePageThokimanContentContainer);
