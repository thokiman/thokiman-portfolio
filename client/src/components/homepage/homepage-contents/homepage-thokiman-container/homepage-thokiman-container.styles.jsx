import styled, { css, keyframes } from 'styled-components';
import { ReactComponent as BlueOutlineThokiman } from '../../../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_blue.svg';
import { ReactComponent as WhiteOutlineThokimanSkeleton } from '../../../../assets/collections-icon/thokiman-icon/page/thokiman_skeleton_white.svg';

const maxWidthSmartphone = 800;
const backgroundColorHomepageThokimanContainer = 'rgba(116, 198, 212, 0.5)';
const lightBlueColor = '#74c6d4';
const getHomepageImage = ({ $rockWaveImage }) => {
  return $rockWaveImage.item.iconPath;
};

const HomepageThokimanContainerIsNotActivated = css`
  display: none;
`;

const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return HomepageThokimanContainerIsNotActivated;
  return null;
};

export const HomepageThokimanContentContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 110vh;
  background-image: url(${getHomepageImage});
  background-size: cover;
`;

export const HomepageThokimanContainer = styled.div`
  height: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${backgroundColorHomepageThokimanContainer};
  width: 96vw;

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    ${getSideBarActive}
  }
`;
export const HomepageThokimanHeadContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 500px;
`;

export const HomepageThokimanBodyContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 10px;
  height: 100%;
  width: 450px;
`;

export const HomepageThokimanHeadOutline = styled(BlueOutlineThokiman)`
  background-size: cover;
  padding: 0 5px;
  background-color: ${lightBlueColor};
  height: 100%;
  width: 100%;
  .thoki-out-blue {
    filter: url(#noise);
  }
`;
const whiteThokimanSkeleton = keyframes`
                             to {
                                 stroke-dashoffset: 0%;
                             }
`;
export const HomepageWhiteThokimanSkeleton = styled(
  WhiteOutlineThokimanSkeleton
)`
  width: 100%;
  height: 100%;

  .thokiman-skeleton-white {
    stroke-miterlimit: 10;
    stroke-dasharray: 700 150;
    stroke-dashoffset: 100%;
    animation: ${whiteThokimanSkeleton} 2s ease-in-out alternate-reverse
      infinite;
  }
`;
HomepageThokimanContentContainer.displayName =
  'HomepageThokimanContentContainer';
HomepageThokimanContainer.displayName = 'HomepageThokimanContainer';
HomepageThokimanHeadContainer.displayName = 'HomepageThokimanHeadContainer';
HomepageThokimanHeadOutline.displayName = 'HomepageThokimanHeadOutline';
HomepageThokimanBodyContainer.displayName = 'HomepageThokimanBodyContainer';
HomepageWhiteThokimanSkeleton.displayName = 'HomepageWhiteThokimanSkeleton';
