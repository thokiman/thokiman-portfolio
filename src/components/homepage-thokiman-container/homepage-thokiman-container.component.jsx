import React from "react";

import {
  HomepageThokimanBodyContainer,
  HomepageThokimanContainer,
  HomepageThokimanContentContainer,
  HomepageThokimanHeadContainer,
  HomepageThokimanHeadOutline,
  HomepageWhiteThokimanSkeleton,
} from "./homepage-thokiman-container.styles";

const HomePageThokimanContentContainer = () => {
  return (
    <HomepageThokimanContentContainer>
      <HomepageThokimanContainer>
        <HomepageThokimanHeadContainer>
          <HomepageThokimanHeadOutline />
        </HomepageThokimanHeadContainer>
        <HomepageThokimanBodyContainer>
          <HomepageWhiteThokimanSkeleton />
        </HomepageThokimanBodyContainer>
      </HomepageThokimanContainer>
    </HomepageThokimanContentContainer>
  );
};

export default HomePageThokimanContentContainer;
