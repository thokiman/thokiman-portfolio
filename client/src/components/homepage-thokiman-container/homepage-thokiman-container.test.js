import React from "react";
import { shallow } from "enzyme";
import { HomePageThokimanContentContainer } from "./homepage-thokiman-container.component";

it("should render HomePageThokimanContentContainer component", () => {
  expect(shallow(<HomePageThokimanContentContainer />)).toMatchSnapshot();
});
