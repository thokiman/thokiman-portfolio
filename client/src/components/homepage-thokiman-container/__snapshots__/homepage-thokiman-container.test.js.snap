// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePageThokimanContentContainer component 1`] = `ShallowWrapper {}`;
